packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -1275,6 +1276,49 @@ class PostsRepo extends AbstractRepo<"Posts"> {
     });
   }
 
+  async fetchEAFundsPosts(userSlugs: string[]) {
+    const eaFundsTagSlug = "effective-altruism-funds";
+    return this.getRawDb().any(`
+      -- PostsRepo.fetchEAFundsPosts
+      WITH "splitAuthors" AS (
+        SELECT p."_id" "postId", coauthor->>'userId' "userId"
+        FROM "Posts" p,
+          UNNEST(
+            ARRAY[JSONB_BUILD_OBJECT('userId', p."userId")] ||
+            p."coauthorStatuses"
+          ) AS coauthor
+        JOIN "Tags" t ON t."slug" = $2
+        WHERE (p."tagRelevance"->t."_id")::INTEGER >= 1
+        AND ${getViewablePostsSelector("p")}
+      ), authors AS (
+        SELECT
+          "splitAuthors"."postId",
+          ARRAY_AGG(u."slug") "slugs",
+          ARRAY_AGG(
+            JSONB_BUILD_OBJECT(
+              '_id', u."_id",
+              'displayName', u."displayName",
+              'pageUrl', $1 || u."slug" || '?utm_source=eafunds'
+            )
+          ) "users"
+        FROM "splitAuthors"
+        JOIN "Users" u ON u."_id" = "splitAuthors"."userId"
+        GROUP BY "splitAuthors"."postId"
+      )
+      SELECT
+        p."_id",
+        p."title",
+        $1 || p."_id" || '/' || p."slug" || '?utm_source=eafunds' "pageUrl",
```

Comment:
This is currently missing the `/posts/` bit of the url (e.g. I'm getting "http://localhost:3000/zZ2vq7YEckpunrQS4/long-term-future-fund-april-2023-grant-recommendations?utm_source=eafunds" locally)

You could fix it like this:
```suggestion
        $1 || 'posts/' || p."_id" || '/' || p."slug" || '?utm_source=eafunds' "pageUrl",
```
Though I think using `postGetPageUrl` would be better in case of future changes

Reply:
Gah - I think this is a copy/paste error after I wrote the query in a SQL editor 🫠